oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java

```diff
@@ -205,39 +205,33 @@ public boolean close() throws IOException {
         return updatesMap.containsValue(Boolean.TRUE);
     }
 
-    private class OakBulkProcessorListener implements BulkProcessor.Listener {
+    private class OAKBulkListener implements BulkListener<String> {
 
         @Override
-        public void beforeBulk(long executionId, BulkRequest bulkRequest) {
+        public void beforeBulk(long executionId, BulkRequest request, List<String> contexts) {
             // register new bulk party
             phaser.register();
 
             // init update status
             updatesMap.put(executionId, Boolean.FALSE);
 
-            bulkRequest.timeout(TimeValue.timeValueMinutes(2));
-
-            LOG.debug("Sending bulk with id {} -> {}", executionId, bulkRequest.getDescription());
+            LOG.debug("Sending bulk with id {} -> {}", executionId, contexts);
             if (LOG.isTraceEnabled()) {
-                LOG.trace("Bulk Requests: \n{}", bulkRequest.requests()
+                LOG.trace("Bulk Requests: \n{}", request.operations()
                         .stream()
-                        .map(DocWriteRequest::toString)
+                        .map(BulkOperation::toString)
                         .collect(Collectors.joining("\n"))
                 );
             }
         }
 
         @Override
-        public void afterBulk(long executionId, BulkRequest bulkRequest, BulkResponse bulkResponse) {
-            LOG.debug("Bulk with id {} processed with status {} in {}", executionId, bulkResponse.status(), bulkResponse.getTook());
+        public void afterBulk(long executionId, BulkRequest request, List<String> contexts, co.elastic.clients.elasticsearch.core.BulkResponse response) {
```

Comment:
Why the full package name for `BulkResponse`?

Reply:
leftover from the migration since both clients contain a `BulkResponse` class.